x-pack/plugins/fleet/public/applications/fleet/index.tsx

```diff
@@ -3,41 +3,18 @@
  * or more contributor license agreements. Licensed under the Elastic License;
  * you may not use this file except in compliance with the Elastic License.
  */
-import React, { memo, useEffect, useState } from 'react';
+import React from 'react';
 import ReactDOM from 'react-dom';
-import useObservable from 'react-use/lib/useObservable';
-import { HashRouter as Router, Redirect, Switch, Route, RouteProps } from 'react-router-dom';
-import { i18n } from '@kbn/i18n';
-import { FormattedMessage } from '@kbn/i18n/react';
-import styled from 'styled-components';
-import { EuiErrorBoundary, EuiPanel, EuiEmptyPrompt, EuiCode } from '@elastic/eui';
+import { Redirect, Route, RouteProps } from 'react-router-dom';
 import { CoreStart, AppMountParameters } from 'src/core/public';
-import { KibanaContextProvider } from '../../../../../../src/plugins/kibana_react/public';
-import { EuiThemeProvider } from '../../../../xpack_legacy/common';
 import {
   IngestManagerSetupDeps,
   IngestManagerConfigType,
   IngestManagerStartDeps,
 } from '../../plugin';
-import { PAGE_ROUTING_PATHS } from './constants';
-import { DefaultLayout, WithoutHeaderLayout } from './layouts';
-import { Loading, Error } from './components';
-import { IngestManagerOverview, EPMApp, AgentPolicyApp, FleetApp, DataStreamApp } from './sections';
-import {
-  DepsContext,
-  ConfigContext,
-  useConfig,
-  useCore,
-  sendSetup,
-  sendGetPermissionsCheck,
-  licenseService,
-  KibanaVersionContext,
-} from './hooks';
-import { PackageInstallProvider } from './sections/epm/hooks';
-import { FleetStatusProvider, useBreadcrumbs } from './hooks';
-import { IntraAppStateProvider } from './hooks/use_intra_app_state';
+import { licenseService } from './hooks';
 import { UIExtensionsStorage } from './types';
-import { UIExtensionsContext } from './hooks/use_ui_extension';
+import { AppRoutes, FleetAppContext, WithPermissionsAndSetup } from './app';
```

Comment:
👏 for pulling this out to new file. Goodbye 20+ lines of `import`s and 200 lines of JSX